rslib/src/error/windows.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use snafu::Snafu;
+
+use super::AnkiError;
+
+#[derive(Debug, PartialEq, Snafu)]
+#[snafu(visibility(pub))]
+pub struct WindowsError {
+    details: WindowsErrorDetails,
+    source: windows::core::Error,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum WindowsErrorDetails {
+    SettingVoice(windows::Media::SpeechSynthesis::VoiceInformation),
+    SettingRate(f32),
+    Synthesizing(String),
```

Comment:
To avoid giant errors, we probably want to truncate the included text to a maximum of something like 40 characters, or just omit it (without using SSML, I would be surprised if the content of the text had any influence over whether the request failed or not).